sky/adaptors/azure.py

```diff
@@ -55,10 +59,131 @@ def get_client(name: str, subscription_id: str):
         elif name == 'resource':
             from azure.mgmt.resource import ResourceManagementClient
             return ResourceManagementClient(credential, subscription_id)
+        elif name == 'storage':
+            from azure.mgmt.storage import StorageManagementClient
+            return StorageManagementClient(credential, subscription_id)
+        elif name == 'authorization':
+            from azure.mgmt.authorization import AuthorizationManagementClient
+            return AuthorizationManagementClient(credential, subscription_id)
+        elif name == 'graph':
+            from msgraph import GraphServiceClient
+            return GraphServiceClient(credential)
+        elif name == 'container':
+            # Currently, there is no way to check if the given endpoint url
+            # of a container is private or public. Hence, we approach with
+            # try-except block by first assuming the url is a public container.
+            # If it turns out to be a private container, an error will be
+            # caught and treated as private container.
+            # Reference: https://github.com/Azure/azure-sdk-for-python/issues/35770 # pylint: disable=line-too-long 
+            from azure.storage.blob import ContainerClient
+            container_url = kwargs.pop('container_url', None)
+            assert container_url is not None
+            container_client = ContainerClient.from_container_url(container_url)
+            try:
+                container_client.exists()
+            except exceptions().ClientAuthenticationError as e:
+                # Caught when credential is not provided to the private
+                # container url or wrong container name is provided as the
+                # public container url. We reattempt with credentials assuming
+                # user is using private container with access rights.
+                if 'ErrorCode:NoAuthenticationInformation' in e.message:
```

Comment:
what if we enter the `else` branch of this `if` statement? iiuc we would return a container client that does not exist. Is it expected?

Reply:
Removed the `if` statement since `except exceptions().ClientAuthenticationError` is quite obvious of what's happening. It seems like the additional `if` statement doesn't add any value while being error prone.